Developer survey 2020 (#375)
solita/solita.github.com: _posts/2020-12-10-developer-survey-2020.md

## Trends

Usage of cloud platforms is on the rise and that has an impact on lots of things, like solutions used for infrastructure-as-code or CI/CD. For example, [**Jenkins**](https://www.jenkins.io/) was still the most used CI/CD solution, but its usage had dropped sharply. The usage of solutions provided by the cloud platforms, such as **Code** tools in AWS were up. In infrastructure-as-code tools [**Ansible**](https://www.ansible.com/) was still the most used, but its usage had dropped significantly and [**AWS CDK**](https://aws.amazon.com/cdk/), [**Terraform**](https://www.terraform.io/) etc. had a lot more usage now than two years ago.

TypeScript usage has increased significantly in two years. When asking for generally used programming languages, TypeScript was reported by 35% of the respondents. In 2018 it was only used by 19%. This change also meant that JavaScript, even though still the most widely used language in general, dropped from 84% to 69% of usage.

## Conclusion

> What does our technology usage then tell about us as a company and a community of developers? We are passionate to learn and tinker with shiny new toys. There's a time and place for that, but for unproven and bleeding edge technologies that place is often not in actual projects but in other contexts. To be a professional means to use things that get the job done the best possible way, not to use all the shiny things. This can be seen in how many hyped up technologies of the past years are nowhere to be seen her. Elm might have been cool a few years ago, but what would be the situation in five years on, when the system we built is still in use and developed further?

Reviewer: ```suggestion
This can be seen in how many hyped-up technologies of the past years are nowhere to be seen today. Elm might have been cool a few years ago, but what would be the scenario if five years on the system we built requires further development?
```
